esmvalcore/preprocessor/_other.py

```diff
@@ -38,3 +39,93 @@ def clip(cube, minimum=None, maximum=None):
             raise ValueError("Maximum should be equal or larger than minimum.")
     cube.data = da.clip(cube.core_data(), minimum, maximum)
     return cube
+
+
+def add_cell_measure(cube, fx_variables, check_level):
+    """
+    Load requested fx files, check with CMOR standards and add the
+    fx variables as cell measures in the cube containing the data.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Iris cube with input data.
+    fx_variables: dict
+        Dictionary with fx_variable information.
+    check_level: CheckLevels
+        Level of strictness of the checks.
+
+
+    Returns
+    -------
+    iris.cube.Cube
+        Cube with added cell measures.
+    """
+    from esmvalcore.preprocessor._io import concatenate
+    from esmvalcore.cmor.fix import fix_metadata, fix_data
+    from esmvalcore.cmor.check import cmor_check_metadata, cmor_check_data
+
+    if not fx_variables:
+        return cube
+    fx_cubes = iris.cube.CubeList()
+    for _, fx_info in fx_variables.items():
+        if not fx_info:
+            continue
+        if isinstance(fx_info['filename'], str):
+            fx_info['filename'] = [fx_info['filename']]
+        for fx_file in fx_info['filename']:
+            loaded_cube = iris.load(fx_file)
```

Comment:
It would probably be safer to use
https://github.com/ESMValGroup/ESMValCore/blob/9f603be2d726e99fcf069ad747ddc8a73a24e89a/esmvalcore/preprocessor/_io.py#L104
with
https://github.com/ESMValGroup/ESMValCore/blob/9f603be2d726e99fcf069ad747ddc8a73a24e89a/esmvalcore/preprocessor/_io.py#L89
here, so the behaviour is the same as in the preprocessing chain for any other variable.